src/network/networkd-dhcp-static-lease.c

```diff
@@ -0,0 +1,269 @@
+                        return 0;
+                }
+        }
+
+        r = dhcp_static_lease_new(&static_lease);
+        if (r < 0)
+                return r;
+
+        static_lease->network = network;
+        LIST_APPEND(dhcp_static_leases, network->dhcp_static_leases, static_lease);
+        network->n_dhcp_static_leases++;
+
+        if (filename) {
+                static_lease->section = TAKE_PTR(n);
+                r = hashmap_ensure_allocated(&network->dhcp_static_leases_by_section, &network_config_hash_ops);
+                if (r < 0)
+                        return r;
+
+                r = hashmap_put(network->dhcp_static_leases_by_section, static_lease->section, static_lease);
+                if (r < 0)
+                        return r;
+        }
+
+        *ret = TAKE_PTR(static_lease);
+
+        return 0; 
+}
+
+int config_parse_dhcp_static_lease_address(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue, 
+                void *data,
+                void *userdata) {
+
+        Network *network;
+        DHCPStaticLease *current_lease = NULL;
+        union in_addr_union addr;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(userdata);
+
+        network = userdata;
+
+        r = lease_new_static(network, filename, section_line, &current_lease);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to create a new static lease: %s", rvalue);
+                return 0;
+        }
+
+        if (isempty(rvalue)) {
+                current_lease->address = 0;
+                return 0;
+        }
+
+        r = in_addr_from_string(AF_INET, rvalue, &addr);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse DHCPv4 IPv4 address data, ignoring assignment: %s", rvalue);
+
+                return 0;
+        }
+
+        /* set address */
+        r = dhcp_static_lease_set_address(current_lease, &addr.in);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse DHCPv4 IPv4 address data, ignoring assignment: %s", rvalue);
+                return 0;
+        }
+        return 0;
+}
+
+int config_parse_dhcp_static_lease_hwaddr(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        Network *network;
+        DHCPStaticLease *current_lease = NULL;
```

Comment:
Same here. Please introduce `dhcp_static_lease_free_or_set_invalidp()` or something and use it with `_cleanup_` attribute.